### DISCUSSION TITLE
userEvent for more 'atomic' mouse events like onMouseDown / onMouseUp?

### DISCUSSION THREAD

**Discussion by dinosaurL** [testing-library/user-event#1140] [Q&A]
Since in [your documentation](https://testing-library.com/docs/dom-testing-library/api-events/),
 you advocate using `userEvent` instead of `fireEvent`, 

will you finally introduce more pointer/mouse specific `userEvent` like `mouseDown` / `mouseUp`
instead of having to use e.g. `fireEvent.mouseDown(element)`?

**Comment by VernonCrabtree** [comment]
I have been trying to use user-event.pointer but not getting anywhere.
While LeftMouse does click where I want, RightMouse does not.
While LeftMouse> does seem to trigger holding the mouse down, /LeftMouse seem to be ignored.

It's all a bit hit-and-miss

**Reply by severo** [reply]
@VernonCrabtree do you have a reproducible code example of the issues you encounter?

I had to struggle a bit, but I could control the mouse finely. I share my example, in case it helps others:

```js
const user = userEvent.setup()
await user.pointer([
  // left button down on the element, setting the mouse coordinates
  { keys: '[MouseLeft>]', target: element, coords: { x: 800, y: 0 } },
  // move the mouse left by 50px
  { coords: { x: 750, y: 0 } },
  // release the left button
  { keys: '[/MouseLeft]' },
])
```

**Reply by VernonCrabtree** [reply]
That was a year ago - I guess I found another solution to the issue I was having.
What I notice in your solution is that the mouse down and mouse up are in the same pointer call - I don't think I attempted it this way.
I would expect this to function in separate calls as well as in a single call - it is the same user object after all and tracking these interactions is (?) the function of this object.

Thank you for the example - I will try it this way if I come across the situation again. Would be good if the documentation had this too, but here is probably good enough - it can be found via search after all
